src/com/amazon/ion/impl/IonReaderBinaryIncremental.java

```diff
@@ -0,0 +1,1932 @@
+        private final ImportLocation importLocation;
+
+        SymbolTokenImpl(String text, int sid, ImportLocation importLocation) {
+            this.text = text;
+            this.sid = sid;
+            this.importLocation = importLocation;
+        }
+
+        @Override
+        public String getText() {
+            return text;
+        }
+
+        @Override
+        public String assumeText() {
+            if (text == null) {
+                throw new UnknownSymbolException(sid);
+            }
+            return text;
+        }
+
+        @Override
+        public int getSid() {
+            return sid;
+        }
+
+        // Will be @Override once added to the SymbolToken interface.
+        public ImportLocation getImportLocation() {
+            return importLocation;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("SymbolToken::{text: %s, sid: %d, importLocation: %s}", text, sid, importLocation);
+        }
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof SymbolToken)) return false;
+
+            SymbolToken other = (SymbolToken) o;
+            if(getText() == null || other.getText() == null) {
+                return getText() == other.getText();
+            }
+            return getText().equals(other.getText());
+        }
+
+        @Override
+        public int hashCode() {
+            if(getText() != null) return getText().hashCode();
+            return 0;
+        }
+    }
+
+    /**
+     * Read-only snapshot of the local symbol table at the reader's current position.
+     */
+    private class LocalSymbolTableSnapshot implements SymbolTable {
+
+        // The system symbol table.
+        private final SymbolTable system = SharedSymbolTable.getSystemSymbolTable(majorVersion);
+
+        // The max ID of this local symbol table.
+        private final int maxId;
+
+        // The max ID that maps to a shared symbol table imported by this local symbol table.
+        private final int importedTablesMaxId;
+
+        // The shared symbol tables imported by this local symbol table.
+        private final SymbolTable[] importedTables;
+
+        // Map representation of this symbol table. Keys are symbol text; values are the lowest symbol ID that maps
+        // to that text.
+        final Map<String, Integer> mapView;
+
+        // List representation of this symbol table, indexed by symbol ID.
+        final List<String> listView;
+
+        LocalSymbolTableSnapshot() {
+            maxId = numberOfSymbols - 1;
+            importedTablesMaxId = importMaxId;
+            // Map with initial size the number of symbols and load factor 1, meaning it must be full before growing.
+            // It is not expected to grow.
+            listView = new ArrayList<String>(symbols.subList(0, numberOfSymbols));
+            mapView = new HashMap<String, Integer>(numberOfSymbols, 1);
```

Comment:
The main thing I want to avoid is a guaranteed growth. As an alternative to a load factor of 1, we can use the default but inflate the initial size beyond what we need. Here's what I have:

```java
mapView = new HashMap<String, Integer>((int) Math.ceil(numberOfSymbols / 0.75), 0.75f);
```

Reply:
That works for me. 👍